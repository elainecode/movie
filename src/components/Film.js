import React from 'react'
import { Link } from 'react-router-dom'

const Film = () => {
    return (
        <div>
         <Link to='/films/wwwd'> 
         <p> This is a film. Click me </p>
         </Link>
        </div>
    )

}


export default Film