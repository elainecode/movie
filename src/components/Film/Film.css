.list-item-image {
    height: 300px;
}

.card-content-top {
    display: flex;
    justify-content: space-between;
}
