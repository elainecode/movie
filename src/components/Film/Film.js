import React from 'react';
import { Link } from 'react-router-dom';
import {
  Card,
  CardActionArea,
  CardActions,
  CardMedia,
  CardContent,
  Typography,
} from '@material-ui/core';
import './Film.css';

const Film = ({
  title,
  year,
  id,
  image_id,
  genres,
  clickFilm,
  match,
}) => {
  return (
    <div className="scroll-click-event">
      <Card onClick={clickFilm(id)}>
        <CardActionArea>
          <Link to={`${match.url}/films/${id}`}>
            <CardMedia
              id="list-item-image"
              image={`https://image.tmdb.org/t/p/w300_and_h450_bestv2${image_id}`}
              title={title}
            />
          </Link>
          <CardContent>
            <div className="card-content-top">
              <Typography variant="subtitle2">{title}</Typography>
              <Typography variant="subtitle2">{year}</Typography>
            </div>
            <Typography variant="body2">
              {genres.join(', ')}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    </div>
  );
};

export default Film;
