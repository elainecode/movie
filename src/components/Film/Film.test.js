import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Film from './Film';

describe('<Film/>', () => {
  test('renders the component', () => {
    const clickFilm = jest.fn();
    const match = { url: '/home' };
    const genres = [];
    const id = 5;
    const title = '';
    const component = shallow(
      <Film
        id={id}
        clickFilm={clickFilm}
        match={match}
        genres={genres}
        title={title}
      />,
    );

    expect(component.debug()).toMatchSnapshot();
    
  })
});
