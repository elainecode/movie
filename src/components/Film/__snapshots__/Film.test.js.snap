// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Film/> renders the component 1`] = `
<div
  className="scroll-click-event"
  id={5}
>
  <WithStyles(ForwardRef(Card))>
    <WithStyles(ForwardRef(CardActionArea))>
      <Link
        to="/home/films/5"
      >
        <WithStyles(ForwardRef(CardMedia))
          id="list-item-image"
          image="https://image.tmdb.org/t/p/w300_and_h450_bestv2undefined"
          title=""
        />
      </Link>
      <WithStyles(ForwardRef(CardContent))>
        <div
          className="card-content-top"
        >
          <WithStyles(ForwardRef(Typography))
            variant="subtitle2"
          />
          <WithStyles(ForwardRef(Typography))
            variant="subtitle2"
          >
            <span
              className="year"
            />
          </WithStyles(ForwardRef(Typography))>
        </div>
        <WithStyles(ForwardRef(Typography))
          variant="body2"
        />
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(CardActionArea))>
  </WithStyles(ForwardRef(Card))>
</div>
`;
